### PR TITLE
samples: openthread: Add tcat support.

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit introduces support for Thread commissioning over authenticated TLS feature to cli sample.

### CODE REVIEW

File: samples/openthread/cli/CMakeLists.txt [hunk 1]
```diff
@@ -13,5 +13,5 @@ project(openthread_cli)
 target_sources(app PRIVATE src/main.c)
 # NORDIC SDK APP END
 
-target_sources_ifdef(CONFIG_BT app PRIVATE src/ble.c)
+target_sources_ifdef(CONFIG_CLI_SAMPLE_MULTIPROTOCOL app PRIVATE src/ble.c)
```
Comment (edmont) [inline samples/openthread/cli/CMakeLists.txt:16]: Isn't the Kconfig update with `CLI_SAMPLE_MULTIPROTOCOL` missing?

**Review by edmont — COMMENTED** [review]
nit: there are some typos in the commit message.

File: samples/openthread/cli/src/ble.c [hunk 2]
```diff
@@ -15,6 +15,10 @@
 #define ADV_INT_MIN 0x01e0 /* 300 ms */
 #define ADV_INT_MAX 0x0260 /* 380 ms */
 
+#if defined(CONFIG_OPENTHREAD_BLE_TCAT)
+#error "Multiprotocol example and TCAT functionality do not coexist together."
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/src/ble.c:19]: this error is thrown quite late into the building. How about adding to `CMakeLists.txt` : 
```
if (CONFIG_CLI_SAMPLE_MULTIPROTOCOL AND CONFIG_OPENTHREAD_BLE_TCAT)
        message( FATAL_ERROR "Multiprotocol example and TCAT functionality do not coexist together." )
endif()
```

File: samples/openthread/cli/snippets/tcat/tcat.conf [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2144
+
+# Enable Ble feture set
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply samples/openthread/cli/snippets/tcat/tcat.conf]: fixed.

File: samples/openthread/cli/CMakeLists.txt (hunk 1 above)
Comment (canisLupus1313) [reply samples/openthread/cli/CMakeLists.txt:16]: https://github.com/nrfconnect/sdk-nrf/pull/13391/files#diff-fe1b703e068591e6255c6d243be32d704aabe4f2fd9f280182ad867b639401f7R18

File: samples/openthread/cli/snippets/tcat/tcat.conf (hunk 3 above)
[the remark this answers is not in the document]
Comment (edmont) [reply samples/openthread/cli/snippets/tcat/tcat.conf]: There is still a typo: `feture`.

Also in the commit message and PR description: `indtoduces`, `Tchead`.

File: samples/openthread/cli/sample.yaml [hunk 4]
```diff
@@ -8,7 +8,8 @@ tests:
       nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf52840dk_nrf52840
       nrf52840dongle_nrf52840 nrf21540dk_nrf52840
     tags: ci_build
-    extra_args: SNIPPET="ci;multiprotocol;tcp"
+    extra_args: >
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply samples/openthread/cli/sample.yaml:11]: I fixed the indentation to match the rest of the file, usb was copy paste mistake.

File: samples/openthread/cli/README.rst
[the remark this answers is not in the document]
Comment (wiba-nordic) [reply samples/openthread/cli/README.rst:1]: While the two snippets are next to each other, the best way to express this is probably:
```
  Not compatible with the ``tcat`` snippet.
```
As a second line for the `multiprotocol` snippet, and a corresponding one for the `tcat` snippet. Better to mention twice, just to be sure.

File: samples/openthread/cli/src/ble.c (hunk 2 above)
Comment (wiba-nordic) [reply samples/openthread/cli/src/ble.c:19]: I'd suggest "Multiprotocol support and TCAT functionality are not compatible." to match the note that should be added to the readme.

**Comment by wiba-nordic** [comment]
Also needs an entry in the changelog.

File: samples/openthread/cli/src/ble.c (hunk 2 above)
Comment (canisLupus1313) [reply samples/openthread/cli/src/ble.c]: fixed.

File: samples/openthread/cli/README.rst
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply samples/openthread/cli/README.rst:1]: Added comment as @wiba-nordic suggested.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 5]
```diff
@@ -392,6 +392,8 @@ Thread samples
 
   All mentioned features are still available in the master variant.
 
+* Added experimental Thread Over Authenticated TLS implementation.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:395]:
```suggestion
* Added experimental support for Thread Over Authenticated TLS.
```
Generally how we word things being experimental.

File: samples/openthread/cli/README.rst [hunk 6]
```diff
@@ -98,7 +98,8 @@ The following snippets are available:
   For additional options, refer to :ref:`RTT logging <ug_logging_backends_rtt>`.
 * ``debug`` - Enables debugging the Thread sample with GDB thread awareness.
 * ``ci`` - Disables boot banner and shell prompt.
-* ``multiprotocol`` - Enables Bluetooth LE support in this sample.
+* ``multiprotocol`` - Enables Bluetooth LE support in this sample. Not compatible with the ``tcat`` snippet.
+* ``tcat`` - Enables support for Thread comissioning over authenticated TLS. Not compatible with the ``multiprotocol`` snippet.
```
Comment (wiba-nordic) [inline samples/openthread/cli/README.rst:102]:
```suggestion
* ``multiprotocol`` - Enables Bluetooth LE support in this sample.
  Not compatible with the ``tcat`` snippet.
* ``tcat`` - Enables support for Thread comissioning over authenticated TLS.
  Not compatible with the ``multiprotocol`` snippet.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 5 above)
Comment (canisLupus1313) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: fixed.

File: samples/openthread/cli/README.rst (hunk 6 above)
Comment (canisLupus1313) [reply samples/openthread/cli/README.rst]: fixed.

**Review by wiba-nordic — APPROVED** [review]
Docs look good to me.

File: samples/openthread/cli/CMakeLists.txt [hunk 7]
```diff
@@ -13,5 +13,10 @@ project(openthread_cli)
...
 target_sources_ifdef(CONFIG_CLI_SAMPLE_LOW_POWER app PRIVATE src/low_power.c)
+
+
+if (CONFIG_CLI_SAMPLE_MULTIPROTOCOL AND CONFIG_OPENTHREAD_BLE_TCAT)
+        message( FATAL_ERROR "Multiprotocol support and TCAT functionality are not compatible." )
+endif()
```
Comment (maciejbaczmanski) [inline samples/openthread/cli/CMakeLists.txt:22]:
```suggestion
endif()

```

**Review by maciejbaczmanski — APPROVED** [review]

**Review by pjasinski990 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**